src/providers/tpm_provider/utils.rs

```diff
@@ -84,3 +91,134 @@ pub struct PasswordContext {
+        (EccFamily::SecpR1, 384) => Ok(EllipticCurve::NistP384),
+        (EccFamily::SecpR1, 512) => Ok(EllipticCurve::NistP521),
+        _ => Err(ResponseStatus::PsaErrorNotSupported),
+    }
+}
+
+pub fn pub_key_to_bytes(pub_key: PublicKey, p_byte_size: usize) -> Result<Vec<u8>> {
+    match pub_key {
+        PublicKey::Rsa(key) => picky_asn1_der::to_vec(&RsaPublicKey {
+            modulus: IntegerAsn1::from_unsigned_bytes_be(key),
+            public_exponent: IntegerAsn1::from_signed_bytes_be(PUBLIC_EXPONENT.to_vec()),
+        })
+        .or(Err(ResponseStatus::PsaErrorGenericError)),
+        PublicKey::Ecc { x, y } => {
+            if x.len() != p_byte_size || y.len() != p_byte_size {
+                error!(
+                    "Received ECC public key with invalid size: x - {} bytes; y - {} bytes",
+                    x.len(),
+                    y.len()
+                );
+                return Err(ResponseStatus::PsaErrorGenericError);
+            }
+            Ok(elliptic_curve_point_to_octet_string(x, y))
+        }
+    }
+}
+
+fn elliptic_curve_point_to_octet_string(mut x: Vec<u8>, mut y: Vec<u8>) -> Vec<u8> {
+    let mut octet_string = vec![0x04];
+    octet_string.append(&mut x);
+    octet_string.append(&mut y);
+    octet_string
+}
+
+pub fn signature_data_to_bytes(data: SignatureData, p_byte_size: usize) -> Result<Vec<u8>> {
+    match data {
+        SignatureData::RsaSignature(signature) => Ok(signature),
+        SignatureData::EcdsaSignature { mut r, mut s } => {
+            if r.len() != p_byte_size || s.len() != p_byte_size {
+                error!(
+                    "Received ECC signature with invalid size: r - {} bytes; s - {} bytes",
+                    r.len(),
+                    s.len()
+                );
+                return Err(ResponseStatus::PsaErrorGenericError);
+            }
+
+            let mut signature = vec![];
+            signature.append(&mut r);
+            signature.append(&mut s);
+            Ok(signature)
+        }
+    }
+}
+
+pub fn parsec_to_tpm_signature(data: Vec<u8>, key_attributes: KeyAttributes) -> Result<Signature> {
+    Ok(Signature {
+        scheme: convert_asym_scheme_to_tpm(key_attributes.key_policy.key_algorithm)?,
+        signature: bytes_to_signature_data(data, key_attributes)?,
+    })
+}
+
+fn bytes_to_signature_data(data: Vec<u8>, key_attributes: KeyAttributes) -> Result<SignatureData> {
+    match key_attributes.key_type {
+        KeyType::RsaKeyPair | KeyType::RsaPublicKey => Ok(SignatureData::RsaSignature(data)),
+        KeyType::EccKeyPair { .. } | KeyType::EccPublicKey { .. } => {
+            let p_size = usize::try_from(key_attributes.key_bits / 8).unwrap();
+            if data.len() != p_size * 2 {
+                return Err(ResponseStatus::PsaErrorInvalidArgument);
+            }
+
+            let mut r = data;
+            let s = r.split_off(p_size);
+            Ok(SignatureData::EcdsaSignature { r, s })
+        }
```

Comment:
I assume that this extracts `r` and `s` values from the concatenation `data`? Basically what is described in [our book](https://parallaxsecond.github.io/parsec-book/parsec_client/operations/psa_algorithm.html#asymmetricsignature-algorithm):

> The representation of the signature as a byte string consists of the concatenation of the signature values r and s. Each of r and s is encoded as an N-octet string, where N is the length of the base point of the curve in octets. Each value is represented in big-endian order, with the most significant octet first.

Maybe it would be nice to put a comment to explain that and put some context 😃